sky/backends/cloud_vm_ray_backend.py

```diff
@@ -420,32 +411,44 @@ def add_ray_task(self,
                 f' = {job_run_id!r}')
 
         logger.debug('Added Task with options: '
-                     f'{name_str}{cpu_str}{resources_str}{num_gpus_str}')
+                     f'{cpu_str}{resources_str}{num_gpus_str}')
         self._code += [
             sky_env_vars_dict_str,
             textwrap.dedent(f"""\
         script = {bash_script!r}
         if run_fn is not None:
             script = run_fn({gang_scheduling_id}, gang_scheduling_id_to_ip)
 
-        log_path = os.path.expanduser({log_path!r})
 
         if script is not None:
             sky_env_vars_dict['SKYPILOT_NUM_GPUS_PER_NODE'] = {int(math.ceil(num_gpus))!r}
             # Environment starting with `SKY_` is deprecated.
             sky_env_vars_dict['SKY_NUM_GPUS_PER_NODE'] = {int(math.ceil(num_gpus))!r}
 
             ip = gang_scheduling_id_to_ip[{gang_scheduling_id!r}]
-            sky_env_vars_dict['SKYPILOT_NODE_RANK'] = ip_rank_map[ip]
+            rank = ip_rank_map[ip]
+
+            if len(cluster_ips_map) == 1: # Single-node task on single-node cluter
+                name_str = '{task_name},' if {task_name!r} is not None else 'task,'
+                log_path = os.path.expanduser(os.path.join({log_dir!r}, 'run.log'))
+            else: # Single-node or multi-node task on multi-node cluster
+                idx_in_cluster = cluster_ips_map[ip]
+                if cluster_ips_map[ip] == 0:
+                    node_name = 'head'
+                else:
+                    node_name = f'worker{{idx_in_cluster}}'
+                name_str = f'{{node_name}}, rank={{rank}},'
+                log_path = os.path.expanduser(os.path.join({log_dir!r}, f'{{node_name}}-{{ip}}.log'))
```

Comment:
It may be better to have the rank in the name or have both `ip` and rank, instead of only having the `ip`, because the user can infer infer the IP from the `node_name`, but may not be easy to get the rank information. How about `{{rank}}-{{node_name}}.log`?